gamla/functional_generic.py

```diff
@@ -21,14 +21,21 @@ async def to_awaitable(value):
     return value
 
 
+_get_name_for_composition = toolz.compose_left(
+    reversed, curried.map(lambda f: f.__name__), "_THEN_".join
```

Comment:
why `_THEN_`?

Reply:
To indicate this is the compose-left order